src/babel/processor-di.js

```diff
@@ -4,19 +4,37 @@ function processReference(t, path, locationValue, state) {
   const self = getComponentDeclaration(t, path.scope);
   const bodyPath = path.get('body');
 
+  let shadowsOwnName = false;
+  if(self) {
+    bodyPath.traverse({
```

Comment:
Annoying that we have to traverse and look for declarations.
Could we use something like `bodyPath.scope.getBinding(name)` and check that whenever the binding is different from "self" binding? 
Thinking of the performance implication of a nested traverse, but not sure whenever it is faster

Reply:
frankly we need to scan only top level scope where variables can cause conflict. So you are right here - traverse might be an overpower.